app-modules/service-management/src/Models/ServiceRequest.php

```diff
@@ -203,9 +220,17 @@ public static function executeFromCampaignAction(CampaignAction $action): bool|s
                     'status_id' => $action->data['status_id'],
                     'type_id' => $action->data['type_id'],
                     'priority_id' => $action->data['priority_id'],
-                    'assigned_to_id' => $action->data['assigned_to_id'] ?? null,
                     'created_by_id' => $action->campaign->user->id,
                 ]);
+
+                if ($action->data['assigned_to_id']) {
+                    $request->assignments()->create([
+                        'user_id' => $action->data['assigned_to_id'],
+                        'assigned_by_id' => $action->campaign->user->id,
+                        'assigned_at' => now(),
+                        'status' => ServiceRequestAssignmentStatus::Active,
+                    ]);
+                }
```

Comment:
What is the purpose of this code here? Does this provide a data stopgap between the old way of how this was managed to the new way?

And if so, is this needed if the DBs should be wiped anyways?

Reply:
This is part of the creation of a ServiceRequest from a CampaignAction. This is not a stopgap, but is needed for creating assignments from the Campaign process.